spring-cloud-deployer-cloudfoundry/src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/ApplicationLogAccessor.java

```diff
@@ -54,29 +56,32 @@ public ApplicationLogAccessor(LogCacheClient logCacheClient) {
      * @return String containing the log information or empty string if no entries are available.
      */
     public String getLog(String deploymentId, Duration apiTimeout) {
+        logger.info("getLog:{}:{}", deploymentId, apiTimeout);
         Assert.hasText(deploymentId, "id must have text and not null");
         Assert.notNull(apiTimeout, "apiTimeout must not be null");
         StringBuilder stringBuilder = new StringBuilder();
+        ReadRequest request = ReadRequest.builder().sourceId(deploymentId).build();
+        Assert.notNull(request, "request must not be null");
         List<Log> logs = this.logCacheClient
-                .read(ReadRequest.builder().sourceId(deploymentId).build())
+                .read(request)
                 .flatMapMany(this::responseToEnvelope)
                 .collectList()
                 .block(apiTimeout);
         // if no log exists the result set is null.
         if(logs == null) {
             return "";
         }
-        Base64.Decoder decoder = Base64.getDecoder();
         logs.forEach((log) -> {
-            stringBuilder.append(new String(decoder.decode(log.getPayload())));
+            stringBuilder.append(log.getPayloadAsText());
             stringBuilder.append(System.lineSeparator());
         });
+
         return stringBuilder.toString();
     }
 
     private Flux<Log> responseToEnvelope(ReadResponse response) {
         return Flux.fromIterable(response.getEnvelopes().getBatch())
-                .map(Envelope::getLog)
-                .filter(Objects::nonNull);
+                .filter(envelope -> envelope.getLog() != null)
```

Comment:
These 2 new lines do the same thing as they did before but the previous uses method refs and is cleaner IMO.